sockets/sockets.go

```diff
@@ -26,13 +25,6 @@ func ConfigureTransport(tr *http.Transport, proto, addr string) error {
 		return configureNpipeTransport(tr, proto, addr)
 	default:
 		tr.Proxy = http.ProxyFromEnvironment
-		dialer, err := DialerFromEnvironment(&net.Dialer{
-			Timeout: defaultTimeout,
-		})
-		if err != nil {
-			return err
-		}
-		tr.DialContext = dialer.DialContext
```

Comment:
This seems to have caused a problem for me because now the unix transport is used when I have specified an HTTP endpoint.

Reply:
I'm not following you ...

The code you highlighted here was only used to configure the proxy on Go <1.9, and was removed because it is redundant and no longer necessary in Go >=1.9.

Removal of this code should not impact transport selection.  Transport selection occurs a few lines earlier, and was not modified by this change: https://github.com/docker/go-connections/blob/master/sockets/sockets.go#L17

Maybe you have a proxy configured using a UNIX socket?